store/src/java/com/zimbra/cs/filter/SoapToSieve.java

```diff
@@ -206,6 +240,25 @@ private String handleNest(String baseIndents, NestedRule currentNestedRule) thro
 
         nestedIfBlock.append(baseIndents);
         nestedIfBlock.append("}\n");
+
+        if(currentNestedRule.getElseRules() != null && currentNestedRule.getElseRules().size() > 0){
+            if(isElse) {
+                // elseRule can't have nested elseRule
+                throw ServiceException.INVALID_REQUEST("elseRule can't have nested elseRule", null);
+            }
+            boolean elseFound = false;
+            for (NestedRule elseRule : currentNestedRule.getElseRules()) {
+                if (elseFound) {
+                    // elsif or else MUST only follow an if or elsif
+                    throw ServiceException.INVALID_REQUEST("Else or elsif can't follow else", null);
```

Comment:
Perhaps the "Else or ..." may start with a lower-case letter as similar as the line 247?